libmysql/libmysql.c

```diff
@@ -788,13 +788,15 @@ MYSQL_FIELD *cli_list_fields(MYSQL *mysql)
 MYSQL_RES * STDCALL
 mysql_list_fields(MYSQL *mysql, const char *table, const char *wild)
 {
+  #define TABLE_NAME_SIZE 65535
   MYSQL_RES   *result;
   MYSQL_FIELD *fields;
-  char	     buff[258],*end;
+  // the buffer must be 130 characters bigger than TABLE_NAME_SIZE.
```

Comment:
What are the 130 extra exactly, do you know?
And, even though you add some padding, it is not useful because the table name will still be truncated after TABLE_NAME_SIZE characters (like it was truncated after 128 characters before).
Should we just create a buffer with the max size we decided on?
That, or if 130 is for "wild" then pass the buffer length in the outer strmake.

Reply:
130 is for the rest of the bytes of the COM_FIELD_LIST command packet :)

The buffer size is TABLE_NAME_SIZE (16K) + 130 (for the command bytes)

Reply:
I can add this to the comment, if you want

Reply:
Oh yes, the bytes of the packet and also the wildcard.